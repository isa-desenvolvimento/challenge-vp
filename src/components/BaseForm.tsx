import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
