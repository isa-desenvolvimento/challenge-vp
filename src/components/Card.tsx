import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'
import React from 'react'

export const Card = ({
  header,
  content,
  footer,
  primary,
  display,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer display={display}>
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
