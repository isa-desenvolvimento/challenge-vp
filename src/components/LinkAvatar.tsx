import {
  StyleNavAvatar,
  StyleImage,
  StyleTitleDescription,
} from '@/styles/LinkAvatar'
import { LinkAvatarProps } from '@/types/global'
import React from 'react'

export const LinkAvatar = ({
  title,
  description,
  src,
  alt,
  width,
  height,
}: LinkAvatarProps) => {
  return (
    <StyleNavAvatar width={width} role="link-avatar">
      <StyleImage src={src} alt={alt} width={width} height={height} />
      <StyleTitleDescription title={title} description={description} />
    </StyleNavAvatar>
  )
}
