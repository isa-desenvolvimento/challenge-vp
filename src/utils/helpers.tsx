import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
