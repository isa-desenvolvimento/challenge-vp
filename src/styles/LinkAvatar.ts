import { WidthProps } from '@/types/global'
import styled from 'styled-components'
import Image from 'next/image'
import { TitleDescription } from '@/components/TitleDescription'

export const StyleNavAvatar = styled.div.attrs((props: WidthProps) => ({
  width: props.width,
}))<WidthProps>`
  background-color: transparent;
  box-sizing: border-box;
  border-left: 1px solid ${({ theme }) => theme.pallet.gray};
  margin-left: auto;

  padding: 1rem;
  height: 75%;
  align-items: center;
  display: flex;
`

export const StyleImage = styled(Image)`
  display: block;
  border-radius: 50%;
`

export const StyleTitleDescription = styled(TitleDescription)`
  margin: 1rem;
`
