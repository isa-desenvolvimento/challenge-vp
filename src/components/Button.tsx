import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
