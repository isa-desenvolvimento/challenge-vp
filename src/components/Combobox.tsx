import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
