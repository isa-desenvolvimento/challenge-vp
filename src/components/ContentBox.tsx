import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
