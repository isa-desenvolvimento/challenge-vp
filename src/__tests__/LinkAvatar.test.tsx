import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { LinkAvatar } from '@/components/LinkAvatar'

jest.mock('next/image', () => ({
  // eslint-disable-next-line @typescript-eslint/naming-convention
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text
    return <img {...props} />
  },
}))

describe('<LinkAvatar />', () => {
  beforeEach(() =>
    renderWithProvider(
      <LinkAvatar
        title="Link"
        description="Description"
        src="/avatar.png"
        alt="alt"
        width="50"
        height="50"
      />
    )
  )

  it('should render text LinkAvatar', () => {
    expect(screen.getByRole(/link-avatar/i).textContent).toEqual(
      'LinkDescription'
    )
  })

  it('should render snapshot', () => {
    expect(screen.getByRole(/link-avatar/i)).toMatchSnapshot()
  })
})
