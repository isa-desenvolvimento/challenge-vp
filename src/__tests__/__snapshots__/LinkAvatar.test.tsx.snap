// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkAvatar /> should render snapshot 1`] = `
.c2 {
  text-align: justify;
  margin: 1rem;
  display: grid;
  grid-gap: 0.3rem;
}

.c3 {
  color: #60647C;
  font-weight: 400;
}

.c4 {
  color: #b0b3c0;
  font-size: 0.8rem;
}

.c0 {
  background-color: transparent;
  box-sizing: border-box;
  border-left: 1px solid #b0b3c0;
  margin-left: auto;
  padding: 1rem;
  height: 75%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: block;
  border-radius: 50%;
}

<div
  class="c0"
  role="link-avatar"
  width="50"
>
  <img
    alt="alt"
    class="c1"
    height="50"
    src="/avatar.png"
    width="50"
  />
  <div
    class="c2"
  >
    <h5
      class="c3"
    >
      Link
    </h5>
    <span
      class="c4"
    >
      Description
    </span>
    <span
      class="c4"
    />
  </div>
</div>
`;
